docs/extend/extend-apiml/api-mediation-oidc-authentication.md

```diff
@@ -0,0 +1,271 @@
+
+  **Tip:** Consult your OIDC provider documentation for options and requirements available for your type of client application. 
+
+- Users have been assigned to the Client Application.
+
+  To access mainframe resources, users with a distributed authentication must either be directly assigned by the OIDC provider to the client application, or must be part of group which is allowed to work with the client application.     
+
+### ESM configuration 
+Execute the following two tasks to ensure that ESM is properly configured for Identity Mapping of distributed to mainframe user identity.
+
+1. Distributed users identity configuration
+
+The users' identity mappings are defined as distributed user identity filters, which are maintained by the System Authorization Facility (SAF) / External Security Manager (ESM).
+A distributed identity consists of two parts: a distributed identity name, and a trusted registry which governs that identity. 
+API ML provides a Zowe CLI plugin to help administrators to generate a JCL for creating the mapping filter specific for the ESM installed on the target mainframe system. 
+
+  See the [Identity Federation cli plugin](../../user-guide/cli-idfplugin.md)
+
+Alternatively, administrators can use the installed ESM functionality to create, delete, list, and query a distributed identity filter/s:
+ - For RACF consult [RACMAP command](https://www.ibm.com/docs/en/zos/2.3.0?topic=rcs-racmap-create-delete-list-query-distributed-identity-filter).
+ - For CA Top Secret use the [IDMAP Keyword - Implement z/OS Identity Propagation Mapping](https://techdocs.broadcom.com/us/en/ca-mainframe-software/security/ca-top-secret-for-z-os/16-0/administrating/issuing-commands-to-communicate-administrative-requirements/keywords/idmap-keyword-implement-z-os-identity-propagation-mapping.html).
+ - For CA ACF2 use [IDMAP User Profile Data Records](https://techdocs.broadcom.com/us/en/ca-mainframe-software/security/ca-acf2-for-z-os/16-0/administrating/administer-records/user-profile-records/idmap-user-profile-records.html).
+
+2. External Mapper user's permissions.
+  The External Mapper functionality is executed on behalf of the calling `externalMapperUser`. By default, this will be the ZOWE runtime user `ZWESVUSR`.   
+  For the identity mapping calls to succeed, the externalMapperUser needs special privileges which may vary per installed ESM type.
+
+  Perform following validations and configuration steps to ensure that externalMapperUser is properly configured:   
+1. Make sure that the user has a default user group set. The default group could be any existing group.   
+   
+  - For Top Secret, execute the following commands to set default group in the externalMapperUser profile:
+   
+    ```
+    TSS ADD(user_id) GROUP(group_id)
+    TSS ADD(user_id) DFLTGRP(group_id)
+    ```
+   
+    **TIP:** If the externalMapperUser is also used to call z/OSMF, make sure that the user is in IZUUSER group. 
```

Comment:
The externalMapperUser must be able to login (with a passticket) to the z/OSMF (that's why it must be in one of the IZU* groups) and in order to do that it must have a password set (i.e. it must not have NOPASSWORD property in his security profile) .... it is the same as for x509 setup.

Reply:
Does it apply to all ESMs or only to TSS?

Reply:
Yes, it applies to all ESM's ... see https://github.com/zowe/docs-site/blob/d59ed6163b6a57af60a073ea105a5eb7e4e65be2/docs/extend/extend-apiml/authentication-for-apiml-services.md?plain=1#L127

Reply:
This is Outdated.